### DISCUSSION TITLE
Linear algebra error for conn_te() - highly correlated channels?

### DISCUSSION THREAD

**Discussion by ErnestJohnston** [brainets/frites#61] [Q&A] [answered]
I'm using conn_te() to calculate the transfer entropy between pairs of intracranial EEG channels. However, for certain pairs, the function will throw a linear algebra error, "Matrix is not positive definite". I also get the same error with conn_covgc().

I suspect that highly correlated EEG channels can produce this error due how transfer entropy is computed in FRITES (matrix inversion?). This seems like a major issue to me, because highly correlated channels may be reflecting physiological information transfer -- such channel pairs would throw an error when using this implementation of transfer entropy.

Is there something missing in my interpretation? Any thoughts are appreciated!

**Comment by EtienneCmb** [comment]
Hi @ErnestJohnston,

Yes, indeed, that's one possibility. One idea to test this hypothesis could be to add some random noise to your data to "decorrelate" the channels. If the `conn_te` is correctly computed, then you're right, the cholesky decomposition failed because of the highly correlated channels. If the error persists it means that the error is different.

- Do you provide the time vector? What is the time lag you defined?
- How many trials do you have? More generally, what is the shape or your input? 
- If channels are highly correlated, then you should not expect any information transfer between them?

**Reply by ErnestJohnston** [reply]
Hi @EtienneCmb 

The input data is an xarray DataArray of shape (trials: 122, channels: 177, time: 3072) with the time vector having a range of [-2, 4].
I added noise to the data, and the pairs throwing errors have resolved completely.

Regarding your third point, are you saying that highly correlated variables by their nature will be unlikely to have information transfer -- thus, pairs throwing this error are unlikely to have significant results?

**Comment by EtienneCmb** [comment, marked as answer]
Highly correlated time-series are very unlikely to have information transfer between them yes. If possible, I would suggest to drop them.

Best,
